### PR TITLE
Changes to support async `ResolveClient` within `IClientResolveContributor`. #167

### PR DESCRIPTION
**Pull request by junderhill**

### CODE REVIEW

File: src/AspNetCoreRateLimit/Middleware/RateLimitMiddleware.cs [hunk 1]
```diff
@@ -116,7 +116,7 @@ public async Task Invoke(HttpContext context)
             await _next.Invoke(context);
         }
 
-        public virtual ClientRequestIdentity ResolveIdentity(HttpContext httpContext)
+        public virtual async Task<ClientRequestIdentity> ResolveIdentity(HttpContext httpContext)
```
Comment (cristipufu) [inline src/AspNetCoreRateLimit/Middleware/RateLimitMiddleware.cs:119]: Can we please rename this method to `ResolveIdentityAsync`?

Comment (junderhill) [reply src/AspNetCoreRateLimit/Middleware/RateLimitMiddleware.cs]: Absolutely, I've pushed this change.

**Comment by cristipufu** [comment]
@junderhill can you please drop those .csproj / gitlab / myget commits and squash the rest so I can merge this to master?

**Comment by junderhill** [comment]
My apologies! I'm glad you noticed them, I pushed it up to our local nuget repo so we could use the changes before they got merged in. I'll get that sorted for you

**Comment by junderhill** [comment]
@cristipufu I've made those changes now and removed my unintended commits from my fork. Sorry again about that!

**Review by cristipufu — APPROVED** [review]

**Pull request merged**